2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+
+  * **Homework Submission**
+
+  Trainees can submit the link to their pull request or Google Doc. This way it'll be easier to track submissions.
+
+  * **Discussion sessions**
+
+  The trainees will have an option to select a time slot out of the available ones. This makes matching easier.
+
+  * **Checkout session**
+
+  If the trainee passes the discussion session, they'll again be able to give preferred choices for checkout sessions.
+
+  * **Training complete**
+
+  Once the training is complete, the trainees can download certificates, and move to the AMY instructor dashboard
+
+### Salient Features
+
+  **In house messaging and feedback**
+
+  At any point of time, the trainee and the relevant instructors will be able to communicate from within the portal. This solves the hassle of unorganised emails.
+
+  **Preferrential selection of discussion and checkout sessions**
+
+  This makes the process flexible, and comfortable.
+
+  **Email notifications**
+
+  Email notifications would be sent in case of important events such as an upcoming discussion session, so that the applicant doesn't miss any deadlines
+
+### Flowchart
+
+Here's a graphic that summarizes the process
+
+![Image](https://docs.google.com/drawings/d/11PIidJrIwYlGbbb5PvyWXpa6r8lG0OVQ7ZJGbefU6VU/pub?w=920&h=1006)
+
+
+## Technical Details
+
+**Backend**
+
+The portal will use **Django** for obvious reasons.
+
+**Database**
+
+As the portal would be a subapp within the AMY project, it will use the same database as AMY. But since the portal, and AMY, in general would be handling a higher load, `PostgreSQL` would be the preferred database. An issue is already raised [here](https://github.com/swcarpentry/amy/issues/716). If this gets addressed, we'll use PostgreSQL. If not, the system can also run on SQLite during development phase.
+
+**Design Framework**
+
+The design is something I want to pay attention to. The trainee dashboard, as well as the admin will be deeply integrated with every stage of instructor training. Since this is something that is to be used regularly, the UI has to be elegant, which vanilla Bootstrap is not.
+
+I propose to either use another theme, or a different framework, such as `MaterializeCSS` for the Trainee Dashboard. Amy can still continue to use Bootstrap for consistency.
+
+**Network Optimization**
+
+The static files (js, css) are the ones which take up most of the time loading the page and are expected to change infrequently. So, to make the portal load faster, a `Cache-Control:max-age=3600` header would instruct the browser to retain those files for atleast an hour before checking for new updates. This will make the interface significantly faster even on slower connections.
+
+**Emails and Notifications**
+
+It'll be required to send bulk emails to applicants which, as the portal grows, would require using a bulk email service, such as `Amazon SES`. Setting this up earlier would help solve the issue before any such problem arises.
```

Comment:
Really?  Suppose 100 people are in various stages of training at a time, and each one advances by one stage every two weeks.  That's much less email than I send manually...